---
identifier: "CVE-2021-41205"
identifiers:
- "GHSA-49rx-x2rw-pc6f"
- "CVE-2021-41205"
package_slug: "pypi/tensorflow-gpu"
title: "Out-of-bounds Read"
description: "TensorFlow is an open source platform for machine learning. In affected
  versions the shape inference functions for the `QuantizeAndDequantizeV*` operations
  can trigger a read outside of bounds of heap allocated array. The fix will be included
  in TensorFlow 2.7.0. We will also cherrypick this commit on TensorFlow 2.6.1, TensorFlow
  2.5.2, and TensorFlow 2.4.4, as these are also affected and still in supported range."
date: "2021-11-10"
pubdate: "2021-11-10"
affected_range: ">=2.6.0,<2.6.1||>=2.5.0,<2.5.2||<2.4.4"
fixed_versions:
- "2.6.1"
- "2.4.4"
- "2.4.4"
affected_versions: "All versions starting from 2.6.0 before 2.6.1, all versions starting
  from 2.5.0 before 2.5.2, all versions before 2.4.4"
not_impacted: "All versions before 2.6.0, all versions starting from 2.6.1, all versions
  before 2.5.0, all versions starting from 2.4.4 before 2.5.2"
solution: "Upgrade to versions 2.6.1, 2.4.4, 2.4.4 or above."
urls:
- "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-49rx-x2rw-pc6f"
- "https://nvd.nist.gov/vuln/detail/CVE-2021-41205"
- "https://github.com/tensorflow/tensorflow/commit/7cf73a2274732c9d82af51c2bc2cf90d13cd7e6d"
- "https://github.com/advisories/GHSA-49rx-x2rw-pc6f"
cvss_v2: "AV:L/AC:L/Au:N/C:P/I:N/A:P"
cvss_v3: "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:N/A:H"
uuid: "33c8ee40-80d6-43cb-bde0-08747008b969"
cwe_ids:
- "CWE-1035"
- "CWE-125"
- "CWE-937"
